### PR TITLE
feat(env): per-key b.pin annotation (#125 phase 2)

### PR DESCRIPTION
**Pull request by fentas**
## Summary
- Adds the \`b.pin: true\` annotation from #125 phase 2
- Pinned map nodes in the consumer's local YAML are preserved verbatim across syncs
- Works regardless of strategy (\`replace\` / \`merge\` / \`client\`) or safety tier
- Pinned-but-deleted-upstream keys are restored

## Why
Phase 2 of issue #125: scalpel-level overrides for individual keys without flipping the whole file to \`strategy: client\`. Lets a consumer pin \`kubectl: v1.30\` for a specific reason while still letting \`kustomize\` flow through normal sync.

## Behavior
\`\`\`yaml
binaries:
  kubectl:
    version: v1.30.0
    b.pin: true       # ignore upstream for this entry
  kustomize:
    version: v5.0.0   # no annotation → updated from upstream
\`\`\`

- Truthy: \`true\`, \`yes\`, \`on\` (case-insensitive). \`b.pin: false\` is a no-op so users can toggle it.
- Annotation itself is preserved across syncs.
- Pinned key absent in upstream → re-added from local.
- YAML only; JSON pinning is a follow-up.

## Test plan
- [x] \`go test ./pkg/env/... -run Pin\`
- [x] full \`go test ./...\`
- [x] \`golangci-lint run ./pkg/env/...\`

Refs #125

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/env/env.go [hunk 1]
```diff
@@ -367,6 +382,10 @@ func SyncEnv(cfg EnvConfig, projectRoot, cacheRoot string, lockEntry *lock.EnvEn
 			if err != nil {
 				return nil, err
 			}
+			target, err = applyPins(target)
+			if err != nil {
+				return nil, fmt.Errorf("applying pins: %w", err)
+			}
```
[the remark this answers is not in the document]
Comment (fentas) [reply pkg/env/env.go]: Fair point — added unit-level coverage for the new edge cases (empty pending, header-only pending, non-mapping pending) in 4d43a79. End-to-end SyncEnv-level integration tests for the pin pipeline are tracked as a follow-up; they need a real git repo fixture which would balloon this PR's scope.

File: pkg/env/pin.go [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+// equality check.
+func yamlNodesStructurallyEqual(a, b *yaml.Node) bool {
+	enc := func(n *yaml.Node) []byte {
+		var buf strings.Builder
+		e := yaml.NewEncoder(&yamlStringWriter{&buf})
+		e.SetIndent(2)
+		if err := e.Encode(n); err != nil {
+			return nil
+		}
+		_ = e.Close()
```
[the remark this answers is not in the document]
Comment (fentas) [reply pkg/env/pin.go]: Addressed in 6bd2a78 — yamlNodesStructurallyEqual now handles Encoder.Close errors. On Close failure the inner helper returns nil, the outer comparison falls through to 'not equal', and the substitution path runs (the safer of the two outcomes since a wrong-equality verdict would either skip a needed substitution or do an unnecessary one).

File: pkg/env/pin.go [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+		return nil
+	}
+	var out []pinnedPath
+	// First pass: is THIS map pinned?
+	for i := 0; i+1 < len(n.Content); i += 2 {
+		k := n.Content[i]
+		v := n.Content[i+1]
+		if k.Value == PinAnnotation && isTrueScalar(v) && len(prefix) > 0 {
+			cp := make([]string, len(prefix))
+			copy(cp, prefix)
```
[the remark this answers is not in the document]
Comment (fentas) [reply pkg/env/pin.go:260]: Addressed in 6423e61 — root-level b.pin is now supported as 'pin the entire document', equivalent to strategy: client for that one file. applyPinsYAML returns local verbatim when it sees a root pin. New TestApplyPinsYAML_RootPinPreservesLocalVerbatim covers it.
